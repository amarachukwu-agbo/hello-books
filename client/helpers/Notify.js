import swal from 'sweetalert';

/**
 * @description class notifies users using alerts
 *
 * @class Notify
 */
class Notify {
  /**
   * @method notifyError
   * @memberof Notify
   * @static
   * @param {string} error
   *
   * @returns {void}
   */
  static notifyError(error) {
    swal('', error, 'error');
  }

  /**
   * @method notifyInfo
   * @memberof Notify
   * @static
   * @param {string} info
   *
   * @returns {void}
   */
  static notifyInfo(info) {
    swal('', info, 'info');
  }

  /**
   * @method notifySuccess
   * @memberof Notify
   * @static
   * @param {string} message
   *
   * @returns {void}
   */
  static notifySuccess(message) {
    swal('', message, 'success');
  }
}

export default Notify;
