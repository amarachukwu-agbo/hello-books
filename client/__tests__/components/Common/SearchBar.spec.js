import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import Notify from '../../../helpers/Notify';
import SearchBar from '../../../components/Common/Searchbar';

let wrapper;
const props = {
  searchBooks: sinon.spy(() => Promise.resolve()),
  isSearching: false,
};

const e = {
  preventDefault: jest.fn(),
};

sinon.spy(SearchBar.prototype, 'searchBook');
const notifyInfo = sinon.spy(Notify, 'notifyInfo');
const notifyError = sinon.spy(Notify, 'notifyError');

describe('<SearchBar />', () => {
  beforeEach(() => {
    wrapper = shallow(<SearchBar {...props} />);
  });

  it('renders correctly', () => {
    wrapper = shallow(<SearchBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles input change', () => {
    wrapper.find('#search-input').simulate('change', {
      target: {
        value: 'A book',
      },
    });
    expect(wrapper.state().searchParam).toBe('A book');
  });

  it('handles dropdown change', () => {
    wrapper.find('#drop-down-list').simulate('change', 'Author');
    expect(wrapper.state().searchBy).toBe('Author');
  });

  it('handles submission of the search form', () => {
    wrapper.setState({
      searchBy: 'Author',
      searchParam: 'Ken Follet',
    });
    wrapper.find('form').simulate('submit', e);
    expect(SearchBar.prototype.searchBook.called).toEqual(true);
    expect(props.searchBooks.called).toEqual(true);
    expect();
  });

  it('notifies user when search criteria is not chosen', () => {
    wrapper.setState({
      searchBy: 'Search By',
      searchParam: 'Romance',
    });
    wrapper.instance().searchBook();
    expect(notifyInfo.called).toEqual(true);
    notifyInfo.restore();
  });

  it('notifies user when the search input is empty', () => {
    wrapper.setState({
      searchParam: '',
      searchBy: 'Author',
    });
    wrapper.instance().searchBook();
    expect(notifyError.called).toEqual(true);
  });
});
