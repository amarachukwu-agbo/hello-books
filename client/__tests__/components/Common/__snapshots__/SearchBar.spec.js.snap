// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    isSearching={false}
    searchBooks={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container search"
      >
        <div
          className="row"
        >
          <div
            className="col s12 search-div"
          >
            <h5>
              Search Books
            </h5>
            <div
              className="row"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="col s4 m3 l2"
                >
                  <div
                    className="input-field"
                  >
                    <Uncontrolled(DropdownList)
                      data={
                        Array [
                          "Title",
                          "Author",
                          "Subject",
                        ]
                      }
                      disabled={false}
                      id="drop-down-list"
                      onChange={[Function]}
                      value="Search By"
                    />
                  </div>
                </div>
                <div
                  className="col s8 m9 l10"
                >
                  <div
                    className="input-field"
                  >
                    <input
                      className="white grey-text text-darken-3"
                      disabled={false}
                      id="search-input"
                      onChange={[Function]}
                      placeholder="Book title, author or subject..."
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>,
      "className": "section-search white-text center",
      "id": "search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col s12 search-div"
          >
            <h5>
              Search Books
            </h5>
            <div
              className="row"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="col s4 m3 l2"
                >
                  <div
                    className="input-field"
                  >
                    <Uncontrolled(DropdownList)
                      data={
                        Array [
                          "Title",
                          "Author",
                          "Subject",
                        ]
                      }
                      disabled={false}
                      id="drop-down-list"
                      onChange={[Function]}
                      value="Search By"
                    />
                  </div>
                </div>
                <div
                  className="col s8 m9 l10"
                >
                  <div
                    className="input-field"
                  >
                    <input
                      className="white grey-text text-darken-3"
                      disabled={false}
                      id="search-input"
                      onChange={[Function]}
                      placeholder="Book title, author or subject..."
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>,
        "className": "container search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col s12 search-div"
          >
            <h5>
              Search Books
            </h5>
            <div
              className="row"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="col s4 m3 l2"
                >
                  <div
                    className="input-field"
                  >
                    <Uncontrolled(DropdownList)
                      data={
                        Array [
                          "Title",
                          "Author",
                          "Subject",
                        ]
                      }
                      disabled={false}
                      id="drop-down-list"
                      onChange={[Function]}
                      value="Search By"
                    />
                  </div>
                </div>
                <div
                  className="col s8 m9 l10"
                >
                  <div
                    className="input-field"
                  >
                    <input
                      className="white grey-text text-darken-3"
                      disabled={false}
                      id="search-input"
                      onChange={[Function]}
                      placeholder="Book title, author or subject..."
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </form>
            </div>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5>
                Search Books
              </h5>,
              <div
                className="row"
              >
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="col s4 m3 l2"
                  >
                    <div
                      className="input-field"
                    >
                      <Uncontrolled(DropdownList)
                        data={
                          Array [
                            "Title",
                            "Author",
                            "Subject",
                          ]
                        }
                        disabled={false}
                        id="drop-down-list"
                        onChange={[Function]}
                        value="Search By"
                      />
                    </div>
                  </div>
                  <div
                    className="col s8 m9 l10"
                  >
                    <div
                      className="input-field"
                    >
                      <input
                        className="white grey-text text-darken-3"
                        disabled={false}
                        id="search-input"
                        onChange={[Function]}
                        placeholder="Book title, author or subject..."
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>
              </div>,
            ],
            "className": "col s12 search-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search Books",
              },
              "ref": null,
              "rendered": "Search Books",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  onSubmit={[Function]}
                >
                  <div
                    className="col s4 m3 l2"
                  >
                    <div
                      className="input-field"
                    >
                      <Uncontrolled(DropdownList)
                        data={
                          Array [
                            "Title",
                            "Author",
                            "Subject",
                          ]
                        }
                        disabled={false}
                        id="drop-down-list"
                        onChange={[Function]}
                        value="Search By"
                      />
                    </div>
                  </div>
                  <div
                    className="col s8 m9 l10"
                  >
                    <div
                      className="input-field"
                    >
                      <input
                        className="white grey-text text-darken-3"
                        disabled={false}
                        id="search-input"
                        onChange={[Function]}
                        placeholder="Book title, author or subject..."
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="col s4 m3 l2"
                    >
                      <div
                        className="input-field"
                      >
                        <Uncontrolled(DropdownList)
                          data={
                            Array [
                              "Title",
                              "Author",
                              "Subject",
                            ]
                          }
                          disabled={false}
                          id="drop-down-list"
                          onChange={[Function]}
                          value="Search By"
                        />
                      </div>
                    </div>,
                    <div
                      className="col s8 m9 l10"
                    >
                      <div
                        className="input-field"
                      >
                        <input
                          className="white grey-text text-darken-3"
                          disabled={false}
                          id="search-input"
                          onChange={[Function]}
                          placeholder="Book title, author or subject..."
                          type="text"
                          value=""
                        />
                      </div>
                    </div>,
                  ],
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <div
                        className="input-field"
                      >
                        <Uncontrolled(DropdownList)
                          data={
                            Array [
                              "Title",
                              "Author",
                              "Subject",
                            ]
                          }
                          disabled={false}
                          id="drop-down-list"
                          onChange={[Function]}
                          value="Search By"
                        />
                      </div>,
                      "className": "col s4 m3 l2",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Uncontrolled(DropdownList)
                          data={
                            Array [
                              "Title",
                              "Author",
                              "Subject",
                            ]
                          }
                          disabled={false}
                          id="drop-down-list"
                          onChange={[Function]}
                          value="Search By"
                        />,
                        "className": "input-field",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "data": Array [
                            "Title",
                            "Author",
                            "Subject",
                          ],
                          "disabled": false,
                          "id": "drop-down-list",
                          "onChange": [Function],
                          "value": "Search By",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": "div",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <div
                        className="input-field"
                      >
                        <input
                          className="white grey-text text-darken-3"
                          disabled={false}
                          id="search-input"
                          onChange={[Function]}
                          placeholder="Book title, author or subject..."
                          type="text"
                          value=""
                        />
                      </div>,
                      "className": "col s8 m9 l10",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <input
                          className="white grey-text text-darken-3"
                          disabled={false}
                          id="search-input"
                          onChange={[Function]}
                          placeholder="Book title, author or subject..."
                          type="text"
                          value=""
                        />,
                        "className": "input-field",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "white grey-text text-darken-3",
                          "disabled": false,
                          "id": "search-input",
                          "onChange": [Function],
                          "placeholder": "Book title, author or subject...",
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      "type": "div",
                    },
                    "type": "div",
                  },
                ],
                "type": "form",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container search"
        >
          <div
            className="row"
          >
            <div
              className="col s12 search-div"
            >
              <h5>
                Search Books
              </h5>
              <div
                className="row"
              >
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="col s4 m3 l2"
                  >
                    <div
                      className="input-field"
                    >
                      <Uncontrolled(DropdownList)
                        data={
                          Array [
                            "Title",
                            "Author",
                            "Subject",
                          ]
                        }
                        disabled={false}
                        id="drop-down-list"
                        onChange={[Function]}
                        value="Search By"
                      />
                    </div>
                  </div>
                  <div
                    className="col s8 m9 l10"
                  >
                    <div
                      className="input-field"
                    >
                      <input
                        className="white grey-text text-darken-3"
                        disabled={false}
                        id="search-input"
                        onChange={[Function]}
                        placeholder="Book title, author or subject..."
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>,
        "className": "section-search white-text center",
        "id": "search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col s12 search-div"
            >
              <h5>
                Search Books
              </h5>
              <div
                className="row"
              >
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="col s4 m3 l2"
                  >
                    <div
                      className="input-field"
                    >
                      <Uncontrolled(DropdownList)
                        data={
                          Array [
                            "Title",
                            "Author",
                            "Subject",
                          ]
                        }
                        disabled={false}
                        id="drop-down-list"
                        onChange={[Function]}
                        value="Search By"
                      />
                    </div>
                  </div>
                  <div
                    className="col s8 m9 l10"
                  >
                    <div
                      className="input-field"
                    >
                      <input
                        className="white grey-text text-darken-3"
                        disabled={false}
                        id="search-input"
                        onChange={[Function]}
                        placeholder="Book title, author or subject..."
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>,
          "className": "container search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col s12 search-div"
            >
              <h5>
                Search Books
              </h5>
              <div
                className="row"
              >
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="col s4 m3 l2"
                  >
                    <div
                      className="input-field"
                    >
                      <Uncontrolled(DropdownList)
                        data={
                          Array [
                            "Title",
                            "Author",
                            "Subject",
                          ]
                        }
                        disabled={false}
                        id="drop-down-list"
                        onChange={[Function]}
                        value="Search By"
                      />
                    </div>
                  </div>
                  <div
                    className="col s8 m9 l10"
                  >
                    <div
                      className="input-field"
                    >
                      <input
                        className="white grey-text text-darken-3"
                        disabled={false}
                        id="search-input"
                        onChange={[Function]}
                        placeholder="Book title, author or subject..."
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>
              </div>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5>
                  Search Books
                </h5>,
                <div
                  className="row"
                >
                  <form
                    onSubmit={[Function]}
                  >
                    <div
                      className="col s4 m3 l2"
                    >
                      <div
                        className="input-field"
                      >
                        <Uncontrolled(DropdownList)
                          data={
                            Array [
                              "Title",
                              "Author",
                              "Subject",
                            ]
                          }
                          disabled={false}
                          id="drop-down-list"
                          onChange={[Function]}
                          value="Search By"
                        />
                      </div>
                    </div>
                    <div
                      className="col s8 m9 l10"
                    >
                      <div
                        className="input-field"
                      >
                        <input
                          className="white grey-text text-darken-3"
                          disabled={false}
                          id="search-input"
                          onChange={[Function]}
                          placeholder="Book title, author or subject..."
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </form>
                </div>,
              ],
              "className": "col s12 search-div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Search Books",
                },
                "ref": null,
                "rendered": "Search Books",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <form
                    onSubmit={[Function]}
                  >
                    <div
                      className="col s4 m3 l2"
                    >
                      <div
                        className="input-field"
                      >
                        <Uncontrolled(DropdownList)
                          data={
                            Array [
                              "Title",
                              "Author",
                              "Subject",
                            ]
                          }
                          disabled={false}
                          id="drop-down-list"
                          onChange={[Function]}
                          value="Search By"
                        />
                      </div>
                    </div>
                    <div
                      className="col s8 m9 l10"
                    >
                      <div
                        className="input-field"
                      >
                        <input
                          className="white grey-text text-darken-3"
                          disabled={false}
                          id="search-input"
                          onChange={[Function]}
                          placeholder="Book title, author or subject..."
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </form>,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="col s4 m3 l2"
                      >
                        <div
                          className="input-field"
                        >
                          <Uncontrolled(DropdownList)
                            data={
                              Array [
                                "Title",
                                "Author",
                                "Subject",
                              ]
                            }
                            disabled={false}
                            id="drop-down-list"
                            onChange={[Function]}
                            value="Search By"
                          />
                        </div>
                      </div>,
                      <div
                        className="col s8 m9 l10"
                      >
                        <div
                          className="input-field"
                        >
                          <input
                            className="white grey-text text-darken-3"
                            disabled={false}
                            id="search-input"
                            onChange={[Function]}
                            placeholder="Book title, author or subject..."
                            type="text"
                            value=""
                          />
                        </div>
                      </div>,
                    ],
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <div
                          className="input-field"
                        >
                          <Uncontrolled(DropdownList)
                            data={
                              Array [
                                "Title",
                                "Author",
                                "Subject",
                              ]
                            }
                            disabled={false}
                            id="drop-down-list"
                            onChange={[Function]}
                            value="Search By"
                          />
                        </div>,
                        "className": "col s4 m3 l2",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <Uncontrolled(DropdownList)
                            data={
                              Array [
                                "Title",
                                "Author",
                                "Subject",
                              ]
                            }
                            disabled={false}
                            id="drop-down-list"
                            onChange={[Function]}
                            value="Search By"
                          />,
                          "className": "input-field",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "data": Array [
                              "Title",
                              "Author",
                              "Subject",
                            ],
                            "disabled": false,
                            "id": "drop-down-list",
                            "onChange": [Function],
                            "value": "Search By",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "type": "div",
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <div
                          className="input-field"
                        >
                          <input
                            className="white grey-text text-darken-3"
                            disabled={false}
                            id="search-input"
                            onChange={[Function]}
                            placeholder="Book title, author or subject..."
                            type="text"
                            value=""
                          />
                        </div>,
                        "className": "col s8 m9 l10",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <input
                            className="white grey-text text-darken-3"
                            disabled={false}
                            id="search-input"
                            onChange={[Function]}
                            placeholder="Book title, author or subject..."
                            type="text"
                            value=""
                          />,
                          "className": "input-field",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "white grey-text text-darken-3",
                            "disabled": false,
                            "id": "search-input",
                            "onChange": [Function],
                            "placeholder": "Book title, author or subject...",
                            "type": "text",
                            "value": "",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "input",
                        },
                        "type": "div",
                      },
                      "type": "div",
                    },
                  ],
                  "type": "form",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
